src/network/networkd-network.c

```diff
@@ -229,6 +229,19 @@ int network_verify(Network *network) {
         if (network->ignore_carrier_loss < 0)
                 network->ignore_carrier_loss = network->configure_without_carrier;
 
+        if (!IN_SET(network->activation_policy, ACTIVATION_POLICY_UP, ACTIVATION_POLICY_ALWAYS_UP)) {
+                if (network->required_for_online < 0) {
+                        log_debug("%s: ActivationPolicy= is not up and RequiredForOnline= is not set, "
+                                  "Setting RequiredForOnline=false.", network->filename);
+                        network->required_for_online = false;
+                } else if (network->required_for_online == true)
+                        log_warning("%s: ActivationPolicy= is not up and RequiredForOnline=true, "
+                                    "This may cause a delay at boot.", network->filename);
```

Comment:
I do not think it is necessary to keep `required_for_online = true` in this case, but silently disable `required_for_online` if the activation policy is not up (nor bound)...

Reply:
I think it's OK to log if the user explicitly set RequiredForOnline= and we're ignoring it.

Reply:
are you sure we should override an explicitly-configured RequiredForOnline= setting? I agree it's unlikely to be useful very often, but i could see someone wanting to be able to set the policy to 'down' (or 'manual') but also require it for online.

Maybe for 'down' or 'manual', we should only log if RequiredForOnline=yes (since it's still possible for something external to bring up the interface at boot), but for 'always-down' we should force required_for_online=no?

Reply:
pushed a new version that only forces `required_for_online = true` for the `always-down` case, for `down` and `manual` it just logs a warning, does that look ok?